### PR TITLE
[FIX] binary_fields: Mimetype mismatch with some non base64 binary file

### PR DESCRIPTION
**Pull request by jbm-odoo**
If we pass a non base64 binary file, the function convert_to_column may
not work as expected. Indeed, with b64decode, all characters that are
neither in the normal base-64 alphabet nor the alternative alphabet
are discarded prior to the padding check. The first decoded characters
may not be representative of the original file, so the function
guess_mimetype will return a wrong mimetype. For example, a .xlsx file
will be detected as SVG file and an exception will be raised if the user
isn't admin.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/fields.py [hunk 1]
```diff
@@ -1854,11 +1854,15 @@ def convert_to_column(self, value, record, values=None, validate=True):
...
         if value[:1] in (b'P', 'P'):  # Fast detection of first 6 bits of '<' (0x3C)
-            decoded_value = base64.b64decode(value)
-            # Full mimetype detection
-            if (guess_mimetype(decoded_value).startswith('image/svg') and
-                    not record.env.is_system()):
-                raise UserError(_("Only admins can upload SVG files."))
+            try:
+                decoded_value = base64.b64decode(value, validate=True)
+            except:
+                pass
```
Comment (pedrobaeza) [inline odoo/fields.py]:
```suggestion
                decoded_value = value
```

File: odoo/fields.py [hunk 2]
```diff
@@ -1854,11 +1854,15 @@ def convert_to_column(self, value, record, values=None, validate=True):
...
+            else:
```
Comment (pedrobaeza) [inline odoo/fields.py]: Always do the check with above change

**Comment by jbm-odoo** [comment]
Hello @odony 
Can you review this PR?

**Comment by tivisse** [comment]
Dear @xmo-odoo 

I would like to take a look at this small fix. 

Best Kind and Sincere Regards,

Yannick.

**Comment by odony** [comment]
I discussed the patch with @jbm-odoo, the current version seems sensible enough. Next thing that is needed here is to cover all cases with tests. I'll give it a shot based on some test code I've stashed somewhere.

**Comment by pedrobaeza** [comment]
Any news in this? Is this going to be targeted to 12.0, being the first affected version?

**Comment by odony** [comment]
Superseded by #47382 which was based on this code. Thanks!

**Pull request closed without merging**